app/viewmodels/package.scala

```diff
@@ -695,24 +697,37 @@ package object viewmodels {
     }
   }
 
+  implicit class RichRichGuaranteeTypeSeq(value: Seq[generated.GuaranteeType03]) {
+
+    def normaliseOtherGuaranteeRefs: Seq[generated.GuaranteeType03] =
+      value
+        .groupBy(_.sequenceNumber)
+        .map {
+          (_, list) =>
+            val combinedOtherGuaranteeRefs = {
+              val refs = list.flatMap(_.otherGuaranteeReference)
+              Option.when(refs.nonEmpty)(refs.takeSampleWithoutPeriod)
+            }
+            list.head.copy(otherGuaranteeReference = combinedOtherGuaranteeRefs)
+        }
+        .toSeq
```

Comment:
Just trying to work out what this is doing 😅 I'd have thought the sequence number is unique? Does that not make `.groupBy(_.sequenceNumber)` obsolete?